packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,301 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const ModelManager = require('./modelmanager');
+const ModelUtil = require('./modelutil');
+/**
+ * Internal Decorator Utility Class
+ * @private
+ * @class
+ * @memberof module:concerto-core
+ */
+class DecoratorUtil {
+    /**
+     * Parses the dict data into an array of decorator jsons
+     * @param {Object} jsonData - the model ast for a particular namespace
+     * @param {String} namespace - the namespace for which parsing is to be done
+     * @param {String} DCS_VERSION - the version string
+     * @returns {Array<Object>} - the parsed decorator command set array
+     * @private
+     */
+    static parseNonVocabularyDecorators(jsonData, namespace, DCS_VERSION){
+        const patternToDetermineVocab = /^Term_/i;
+        const dcsObjects = [];
+        jsonData.forEach(obj =>{
```

Comment:
By running this loop here, and also in `parseVocabularies`, you are running this loop twice, instead you can pull that into `parseDecosAndVocabs`, you can be done with one loop. You can call either `parseNonVocabularyDecorators ` or `parseVocabularies` based on the conditions you decide to process. At each loop, both functions use `jsonData` just to run the loop.

If you run the loop at `parseDecosAndVocabs`, you can also pull the loop you run on `decos` in `parseDecosAndVocabs` and run the decision making process there, i.e., you can run the `forEach` loop on `jsonData`, then for each `obj` you can get `decos` and run the `forEach` loop on that as well, then finally, based on `dcs.name === 'Term' || patternToDetermineVocab.test(dcs.name)` condition, you can just pass `obj` and `decos` and other things needed to `parseVocabularies` and `parseNonVocabularyDecorators` functions. This would simplify these functions a lot.